### PR TITLE
Update default database cleaner strategy. 

### PR DESCRIPTION
**Pull request by eschlange**
- Update default strategy from truncation to transaction.
- Added unique strategy for feature tests.
- Update specs to accommodate database cleaner config changes.
- Update participant model to handle missing fields on participant.
- Update specs to remove models between specs.
- Updated participant method and assocaited spec to handle missing events.
- Removed unnecessary patient dash spec (scope already tested in activity spec line 109).
- Wiping table for feature test.
- Destroy existing moods to eliminate intermittent spec errors.

[#118847783]

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -59,7 +59,7 @@
   end
 
   config.before(:each) do
-    DatabaseCleaner.strategy = :truncation
+    DatabaseCleaner.strategy = :transaction
```
Comment (michaelwehrley) [inline spec/rails_helper.rb]: If we are using only transactions now, can we remove `DatabaseCleaner` all together?

File: spec/models/thought_spec.rb [hunk 2]
```diff
@@ -8,6 +8,10 @@
                         email: "e@foo.bar")
   end
 
+  before :each do
+    Thought.destroy_all
```
Comment (michaelwehrley) [inline spec/models/thought_spec.rb]: I was wondering if it is reasonable to update the specs so we don't have to use `destroy_all` before them each.  Is that possible or too much overhead?

File: spec/rails_helper.rb (hunk 1 above)
Comment (eschlange) [inline spec/rails_helper.rb]: When I remove the configuration to stop using truncation for the :js specs, we get a fairly large number of failures. If we remove the configuration, I'm not sure how we'd update that specific strategy.

File: spec/models/thought_spec.rb (hunk 2 above)
Comment (eschlange) [inline spec/models/thought_spec.rb]: Any recommendations on how to do so? I'm not sure how that can be accomplished.

Comment (michaelwehrley) [inline spec/models/thought_spec.rb]: We could make that for another story I guess.  We would have to go through each failing spec in the model spec.

File: spec/rails_helper.rb (hunk 1 above)
Comment (michaelwehrley) [inline spec/rails_helper.rb]: Okay.  Great.  Thanks!

**Comment by michaelwehrley** [comment]
Nice updates.  Thanks for looking at this!

**Comment by michaelwehrley** [comment]
👍

File: spec/models/thought_spec.rb (hunk 2 above)
Comment (eschlange) [inline spec/models/thought_spec.rb]: I'll add a TFD maintenance story to look into this.

**Pull request merged**